File: solutions/python/hamming/1/hamming.py
def distance(strand_a, strand_b):
    if len(strand_a) != len(strand_b):
        raise ValueError("Strands must be of equal length.")
    if not len(strand_a):
        return 0
    return sum([list(strand_a)[ix] != list(strand_b)[ix] for ix in range(len(strand_a))])

File: solutions/ocaml/hamming/1/hamming.ml
type nucleotide = A | C | G | T


let hamming_distance strand1 strand2 =  match ( List.length strand1, List.length strand2 ) with
    | (x, y) when x == y -> List.fold_left (+) 0 ( List.map2 ( fun x y -> if x <> y then 1 else 0 ) strand1 strand2 ) |> Result.ok
    | (0, _) -> Result.error "left strand must not be empty"
    | (_, 0) -> Result.error "right strand must not be empty"
    | _ -> Result.error "left and right strands must be of equal length"
